### PR TITLE
Reduce globals

### PR DESCRIPTION
**Pull request by wrefgtzweve**
This pr aims to reduce the global table pollution by localizing variables that should've been local.

From:
image
image
The tool didn't display all the old globals, but there where more.

To:
image

### CODE REVIEW

File: lua/pac3/editor/client/panels/tree.lua [hunk 1]
```diff
@@ -230,7 +230,6 @@ local function install_drag(node)
 				pace.RecordUndoHistory()
 				self.part:SetParent(child.part)
 				pace.RecordUndoHistory()
-				called = true
```
Comment (thegrb93) [inline lua/pac3/editor/client/panels/tree.lua:233]: Sure this isn't some badly made hack used by the caller function?

Comment (thegrb93) [reply lua/pac3/editor/client/panels/tree.lua:233]: I don't see any other use of this var so maybe not.

File: lua/pac3/editor/client/parts.lua [hunk 2]
```diff
@@ -330,7 +331,7 @@ do -- menu
 				pnl:SetImage(groupData.icon)
 			end
 
-			local trap = false
```
Comment (thegrb93) [inline lua/pac3/editor/client/parts.lua:333]: Why'd this local get moved?

Comment (wrefgtzweve) [reply lua/pac3/editor/client/parts.lua:333]: https://github.com/CapsAdmin/pac3/blob/0fe2506c40ff2e907adffeba230bbad0ae08b1f6/lua/pac3/editor/client/parts.lua#L273

In the original code it was being defined above it, thus creating a global

Comment (thegrb93) [reply lua/pac3/editor/client/parts.lua:333]: Oh ok, I see it now. So the code has been completely broken I think. Probably need to test whatever that trap var is used for.

Comment (thegrb93) [reply lua/pac3/editor/client/parts.lua:333]: The rest of the changes look fine.

Comment (wrefgtzweve) [reply lua/pac3/editor/client/parts.lua:333]: Now that im looking at it, i don't think this code ever runs? `pace.AddRegisteredPartsToMenu` is called by 2 functions `pace.OnPartMenu` and `pace.OnNewPartMenu` don't seem to be called anywhere? what the hell

Comment (wrefgtzweve) [reply lua/pac3/editor/client/parts.lua:333]: `pace.OnCreatePart` `pace.OnPartSelected` `pace.OnVariableChanged` `pace.OnPartMenu` `pace.OnNewPartMenu` `pace.OnHoverPart` don't seem to be used? i must be missing something

Comment (CapsAdmin) [reply lua/pac3/editor/client/parts.lua:333]: > `pace.OnCreatePart` `pace.OnPartSelected` `pace.OnVariableChanged` `pace.OnPartMenu` `pace.OnNewPartMenu` `pace.OnHoverPart` don't seem to be used? i must be missing something

they are used "magically" by pace.RunHook("PartSelected") or something like that

Comment (thegrb93) [reply lua/pac3/editor/client/parts.lua:333]: Probably string concatenation being used to call them `"On"..hookname`, which is dumb and needs to be removed someday if so.

Comment (wrefgtzweve) [reply lua/pac3/editor/client/parts.lua:333]: image
yup

Comment (wrefgtzweve) [reply lua/pac3/editor/client/parts.lua:333]: The `trap` move seems to work fine

**Pull request merged**